snuba/pipeline/settings_delegator.py

```diff
@@ -0,0 +1,70 @@
+from typing import Sequence
+
+from snuba.request.request_settings import RequestSettings
+from snuba.state.rate_limit import RateLimitParameters
+
+
+class RateLimiterDelegate(RequestSettings):
+    """
+    When we run the pipeline delegator, we are running multiple queries
+    at the same time. Generally they are duplicates.
+    Each pipeline should not know anything about the environment it runs
+    into otherwise we would increase the pipeline complexity
+    dramatically.
+
+    Still parallel pipelines on separate clusters should not share the
+    same quota and rate limiter.
+
+    The idea is to provide the query pipeline with delegates over the
+    components they need to execute it that create separate namespaces.
+    Specifically the PipelineDelegator provides a RateLimiterDelegate to
+    the secondary query so that the query pipeline can use a separate
```

Comment:
As far as I can see both the primary and secondary query receive the delegate.

Reply:
Right will fix